types/ccd.ts

```diff
@@ -0,0 +1,212 @@
+export interface CaseField {
+  CaseTypeID: string
+  ID: string
+  Label: string
+  HintText?: string
+  FieldType?: string
+  FieldTypeParameter?: string
+  RegularExpression?: string
+  SecurityClassification: "Public"
+  Min?: number
+  Max?: number
+}
+
+export interface AuthorisationCaseField {
+  CaseTypeId: string
+  CaseFieldID: string
+  UserRole: string
+  CRUD: string
+}
+
+export interface CaseEventToField {
+  CaseTypeID: string
+  CaseEventID: string
+  CaseFieldID: string
+  DisplayContext: 'READONLY' | 'OPTIONAL' | 'MANDATORY'
+  PageID: number
+  PageDisplayOrder: number
+  PageFieldDisplayOrder: number
+  FieldShowCondition?: string | null
```

Comment:
What's the reason for all the "| null"s?

Reply:
`?` just gives us `undefined`, so we're adding `null` to allow us to wipe the value by setting it to null. It's not actually used though so I've removed